### ISSUE TITLE
Non Id Unique fields throwing error from findUniqueOrThrow

### ISSUE THREAD

**Issue by AndrewGalea** [prisma/prisma#21079]
### Bug description

When calling `findUniqueOrThrow` on an `@unique` field other than the primary key `id` I receive the following error:

`thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', query-engine/core/src/response_ir/internal.rs:371:77`

This only started happening when I upgraded to the latest Prisma version (previously, I was using v3.x). Other prisma calls such as `findMany` and `findUniqueOrThrow` on the primary key `id` still work. It is only when I call `findUniqueOrThrow` on unique fields.

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}

type GamesHeroes {
  avatarUrl         String
  banned            Boolean @default(false)
  boundSouls        Int @default(0)
  currentHealth     Int
  defensiveArtefact String
  dropped           Boolean @default(false)
  gods              String[]
  health            Int[]
  level             Int @default(1)
  name              String
  offensiveArtefact String
  urlName           String
}

type GodsAvatar {
  name     String
  url_name String
}

type GodsChampions {
  name     String
  url_name String
}

type GodsDivineGifts {
  effigy_power    GodsDivineGiftsEffigyPower
  sacred_artefact String
  warband_bonus   String
}

type GodsDivineGiftsEffigyPower {
  avatar_bonus String
  description  String
  name         String
}

type HeroesActiveAbilities {
  cost        String
  description String
  name        String
}

type HeroesCombatManoeuvres {
  cost        String
  description String
  name        String
}

type HeroesLevel2 {
  active_abilities        HeroesLevel2ActiveAbilities[]
  combat_manoeuvres       HeroesLevel2CombatManoeuvres[]
  unique_innate_abilities HeroesLevel2UniqueInnateAbilities[]
}

type HeroesLevel2ActiveAbilities {
  cost        String
  description String
  name        String
}

type HeroesLevel2CombatManoeuvres {
  cost        String
  description String
  name        String
}

type HeroesLevel2UniqueInnateAbilities {
  description String
  name        String
}

type HeroesLevel3 {
  active_abilities        HeroesLevel3ActiveAbilities[]
  combat_manoeuvres       HeroesLevel3CombatManoeuvres[]
  unique_innate_abilities HeroesLevel3UniqueInnateAbilities[]
}

type HeroesLevel3ActiveAbilities {
  cost        String
  description String
  name        String
}

type HeroesLevel3CombatManoeuvres {
  cost        String
  description String
  name        String
}

type HeroesLevel3UniqueInnateAbilities {
  description String
  name        String
}

type HeroesUniqueInnateAbilities {
  description String
  name        String
}

type HeroesWeapons {
  cost   String
  crit   String
  glance String
  name   String
  reach  String
  solid  String
  type   String
}

type MonstersCombatManoeuvres {
  cost        String
  description String
  name        String
}

type MonstersUniqueInnateAbilities {
  description String
  name        String
}

type MonstersWeapons {
  cost   String
  crit   String
  glance String
  name   String
  reach  String
  solid  String
  type   String
}

type SummonsCombatManoeuvres {
  cost        String
  description String
  name        String
}

type SummonsUniqueInnateAbilities {
  description String
  name        String
}

type SummonsWeapons {
  cost   String
  crit   String
  glance String
  name   String
  reach  String
  solid  String
  type   String
}

model artefacts {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  avatar_url  String
  category    String
  cost        String
  description String
  name        String @unique
  quote       String
  type        String
  url_name    String @unique
}

model classes {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  description String
  name        String @unique
  tag         String
}

model commonInnateAbilities {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  description String
  name        String @unique
}

model faqs {
  id       String   @id @default(auto()) @map("_id") @db.ObjectId
  answer   String
  question String
  tags     String[]
}

model gameDefinitions {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  description String
  name        String @unique
}

model games {
  id       String        @id @default(auto()) @map("_id") @db.ObjectId
  format   String
  gameDate String
  heroes   GamesHeroes[]
  size     String
  souls    Int @default(0)
}

model gods {
  id           String          @id @default(auto()) @map("_id") @db.ObjectId
  avatar       GodsAvatar
  champions    GodsChampions[]
  description  String
  divine_gifts GodsDivineGifts
  logo         String
  name         String @unique
  traits       String
  url_name     String @unique
}

model heroes {
  id                      String                        @id @default(auto()) @map("_id") @db.ObjectId
  active_abilities        HeroesActiveAbilities[]
  agi                     String
  avatar_url              String
  classes                 String[]
  combat_manoeuvres       HeroesCombatManoeuvres[]
  common_innate_abilities String[]
  gods                    String[]
  health                  String[]
  image_url               String
  level_2                 HeroesLevel2
  level_3                 HeroesLevel3
  mag                     String
  mel                     String
  mov                     String
  name                    String @unique
  quotes                  String[]
  race                    String
  res                     String
  rng                     String
  soul_harvest            String
  title                   String
  unique_innate_abilities HeroesUniqueInnateAbilities[]
  url_name                String @unique
  weapons                 HeroesWeapons[]
}

model monsters {
  id                       String                          @id @default(auto()) @map("_id") @db.ObjectId
  agi                      String
  avatar_url               String
  bounty                   String
  bounty_artefacts         String[]
  combat_manoeuvres        MonstersCombatManoeuvres[]
  common_innate_abilities  String[]
  health                   String
  image_url                String
  mag                      String
  mel                      String
  monster_innate_abilities String[]
  mov                      String
  name                     String @unique
  quotes                   String[]
  race                     String
  res                      String
  rng                      String
  tier                     String
  unique_innate_abilities  MonstersUniqueInnateAbilities[]
  url_name                 String @unique
  weapons                  MonstersWeapons[]
}

model pages {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  body        String
  description String
  name        String @unique
  title       String
}

model races {
  id          String @id @default(auto()) @map("_id") @db.ObjectId
  description String
  logo        String
  name        String @unique
  traits      String
}

model summons {
  id                      String                         @id @default(auto()) @map("_id") @db.ObjectId
  agi                     String
  avatar_url              String
  combat_manoeuvres       SummonsCombatManoeuvres[]
  common_innate_abilities String[]
  health                  String
  image_url               String
  mag                     String
  mel                     String
  mov                     String
  name                    String @unique
  quotes                  String[]
  race                    String
  res                     String
  rng                     String
  summon_cost             String
  title                   String
  unique_innate_abilities SummonsUniqueInnateAbilities[]
  url_name                String @unique
  weapons                 SummonsWeapons[]
}
```

```ts
// Add your code using Prisma Client
import { PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

export default prisma;
```

### Environment & setup

- OS: macOS
- Database: MongoDB
- Node.js version: v18.16.0

### Prisma Version

```
v5.3.0
```

**Comment by AndrewGalea** [comment]
I have learned more about this. The error is only thrown when the `model` has a `composite type`. Entities, such as `artefacts` in my scheme above, work when calling `findUniqueOrThrow` on the unique `name` field because no `composite types` are associated with that entity.

**Comment by AndrewGalea** [comment]
I solved this. I always thought you needed to have the `include` to return the associated (child) records to an entity, but when I removed them, the child records were still returned and the error disappeared.

So, for this example below, it would throw the error. 

`const god = await prisma.gods.findUniqueOrThrow({
            where: {
                url_name: godUrlName,
            },
            include: {
                divine_gifts: true,
                champions: true,
                avatar: true,
            },
        });`

However, the following would not and it worked correctly.

`const god = await prisma.gods.findUniqueOrThrow({
            where: {
                url_name: godUrlName,
            },
        });`

Note, in this example, the child entities `divine_gifts`, `champions` and `avatar` were all returned by the query.
